server/src/middleware.ts

```diff
@@ -1,19 +1,30 @@
-import { NextRequest, NextResponse } from "next/server";
+import { NextRequest, NextResponse } from 'next/server';
+import jwt from 'jsonwebtoken';
 
 export function middleware(request: NextRequest) {
-    console.log("Middleware called");
+    // Skip token validation for API routes
+    if (request.nextUrl.pathname.startsWith('/api/')) {
+        return NextResponse.next();
+    }
 
-    // verify authentication cookie here (JWT or any other method)
-    
     let cookie = request.cookies.get("token");
-    if (!cookie || cookie.value != 'authenticated') {
+    if (!cookie || !isValidToken(cookie.value)) {
         return NextResponse.redirect(`${process.env.BASE_URL}/login`);
     }
     return NextResponse.next();
 }
 
+const isValidToken = (token: string) => {
+    try {
+        const decoded = jwt.verify(token, process.env.JWT_SECRET || 'your-default-secret');
```

Comment:
Let's not assume that `JWT_SECRET` is nullish. It had to be supplied for the application to work. Remove `|| 'your-default-secret'`